lib/spack/spack/ci.py

```diff
@@ -1723,8 +1723,8 @@ def copy_stage_logs_to_artifacts(job_spec, job_log_dir):
 
     stage_dir = job_pkg.stage.path
     tty.debug("stage dir: {0}".format(stage_dir))
-    build_out_src = os.path.join(stage_dir, "spack-build-out.txt")
-    copy_files_to_artifacts(build_out_src, job_log_dir)
+    for file in [job_pkg.log_path, job_pkg.env_mods_path, *job_pkg.builder.archive_files]:
```

Comment:
Won't this also copy the intermediate build logs as well, `spack-build-out-01.txt`, `spack-build-out-02.txt`, etc. I think we explicitly avoid that to prevent the cache from being too large.

Reply:
Not that I know of, but I don't know where those files get created or installed so I can't confirm or deny.

Reply:
Looking at the code in `package_base.py`, I think the `log_path` is just a single file, but it handles older file names that used to house the same content.  Also based on the comments above [this](https://github.com/spack/spack/blob/develop/lib/spack/spack/installer.py#L1983) line, it should be safe to include the `env_mods_path` contents in the artifacts.  

Regarding the addition of `builder.archive_files`, it looks like those vary with build system.  For example, for autotools packages, it will be the `config.log`, and maybe some other log file.  For cmake packages, it will include the `CMakeCache.txt`.  Including these in the artifacts makes sense to me.

@kwryankrattiger Where do you think the numbered (which I think are for the different build phases) would be coming from?

Reply:
Instead of inspecting source, can we just run that CI rerun all thingy from spackbot and test what actually gets copied?

Reply:
That will be expensive, but I guess we can do it.  

This would be a good time to ask spackbot to rebuild everything _for just a specific stack_, but currently it doesn't support that.  Alternatively, maybe there is some other mode of forcing it to rebuild a subset that could be effective but less costly, e.g. asking it to test all build systems, rebuild only roots, rebuild a random sample, etc.  I'm just thinking out loud.  

@tgamblin

Reply:
@adamjstewart can you add a variant to something like `readline` which will trigger a smaller subset of rebuilds, rebuild everything is kind of overkill to figure out which files get uploaded to artifacts.

Reply:
Not sure why CI is failing, I didn't change anything that would prevent readline from building. I built readline locally and all of the listed files are in exactly the location reported as non-existent in CI. It kinda feels like there's something else going on here...